### PR TITLE
feat: add continue button

### PR DESCRIPTION
**Pull request by 09473ZH**
Supported the feature to display 'Continue' while awaiting user input (currently lacking logic for controlling button visibility, to be completed before merging #1487 

---update:
image
image

### CODE REVIEW

**Comment by iFurySt** [comment]
IMHO, the continue behavior can be implicitly embedded in sending a message. it will be more natural for UX. WDYT?

**Comment by xingyaoww** [comment]
@iFurySt Do you think we should directly change the send button into "[continue]" when (1) the agent is awaiting user, (2) the input box is empty? And when user types anything into the input box, the "[continue]" button will go back to what it looks like (upward arrow)?

image

**Comment by iFurySt** [comment]
start a new session/task:
1. send a message (to start a task)
2. (user is waiting for completion, providing info, or continuing task based on prev result)
3. send a message (to provide more info to help the agent complete the task or to continue the next task based on the tasks already completed)

step 2. and step 3. may loop multiple times until the user stops or restarts a new fresh session.

@xingyaoww wdyt?

**Comment by xingyaoww** [comment]
@iFurySt Since https://github.com/OpenDevin/OpenDevin/pull/1290 is merged, we already support awaiting input (e.g., user sends follow-up messages instead of re-start), check demo:

https://github.com/OpenDevin/OpenDevin/assets/38853559/d0239ed5-8521-4df0-b6d4-0ab36f128aa5

This PR just adds a functionality that, when use don't want to manually type "continue" for agent to continue working on the task, click the button would sends a user message "continue" to the agent on user's behalf, similar to OpenAI ChatGPT's continue generation button.

**Review by rbren — CHANGES_REQUESTED** [review]
CC @openartist

I think we need a design pass at this one. Hopefully it can also solve https://github.com/OpenDevin/OpenDevin/issues/1493

**Comment by iFurySt** [comment]
image

i think the @openartist has finished the UI design. @09473ZH is it possible to take some time to refactor it 😄

**Comment by 09473ZH** [comment]
> image
> i think the @openartist has finished the UI design. @09473ZH is it possible to take some time to refactor it 😄

Oh, that's nice!  I wonder why there's a refresh icon next to 'Continue'?

**Comment by rbren** [comment]
@09473ZH good question--maybe it should be more like a forward arrow `->`

**Comment by rbren** [comment]
This is looking great!

Screenshot 2024-05-09 at 1 27 41 PM


A few minor style tweaks:
* Can we get it to float over the message (without obscuring text) like in the designs?
* Can we make it a bit smaller?

**Review by rbren — APPROVED** [review]
Approving as-is, but if feasible would love to tweak the styles per my suggestions above before merging

**Comment by 09473ZH** [comment]
> Approving as-is, but if feasible would love to tweak the styles per my suggestions above before merging


Thanks for catching that oversight. I've updated my code. Let me know if there's anything else I should change. :)

File: frontend/src/components/chat/ChatInput.tsx
Comment (amanape) [inline frontend/src/components/chat/ChatInput.tsx:68]: The `<ChatInput>` component contains logic related to user input for the chat. This code block seems out of place as it has nothing to do with the user input. Maybe you can move this logic to the parent, above `<ChatInput>`? That would also remove the dependency on `currentTaskState`. (imagine you are going to reuse this component in another part of the app, does it make sense to also reuse this "continue" logic you have implemented here?)

Comment (09473ZH) [reply frontend/src/components/chat/ChatInput.tsx:68]: I've considered both situations. At first, I thought the "continue button" was more like a quick reply linked to the `<ChatInput />`. After looking into it more carefully, I agree that the way you mentioned makes more sense :)

File: frontend/src/components/chat/ChatInput.tsx
Comment (09473ZH) [reply frontend/src/components/chat/ChatInput.tsx:64]: Agree! Just to be safe, I went ahead and followed the design draft, but honestly, I think "right arrow" would work better. How about this one? 
image

Comment (amanape) [reply frontend/src/components/chat/ChatInput.tsx]: Looks good!

File: frontend/src/components/ChatInterface.tsx
Comment (09473ZH) [reply frontend/src/components/ChatInterface.tsx:1]: Sry about that, it seems there was an error during my code merge

**Review by amanape — APPROVED** [review]

**Pull request merged**
